streaming/base/storage/upload.py

```diff
@@ -258,6 +258,12 @@ def upload_file(self, filename: str):
 
         @retry(num_attempts=self.retry)
         def _upload_file():
+            extra_args = {}
+            # Check for canned ACL environment variable
+            canned_acl = os.environ.get('S3_CANNED_ACL')
```

Comment:
Canned ACLs (Access Control Lists) are predefined permissions you can apply to S3 buckets and objects. They provide a simple way to control who can access your data without creating explicit IAM policies. Admin must have enabled the ACL policy for an S3 bucket per user or a group of users, and when that same user sets the `S3_CANNED_ACL` environment variable, they can access the data. Otherwise, it will just fail.